frame/support/src/traits.rs

```diff
@@ -1547,6 +1547,61 @@ pub trait EnsureOrigin<OuterOrigin> {
 	fn successful_origin() -> OuterOrigin;
 }
 
+/// Trait implemented for pallet dispatchable type by `decl_module`.
+pub trait PalletDispatchable {
+	/// The origin type of the runtime, (i.e. `frame_system::Trait::Origin`).
+	type Origin;
```

Comment:
```suggestion
	type Origin: OriginTrait;
```

Both work, wonder why not restrict it?

Reply:
actually it breaks some tests but I can fix them

Reply:
I uncommitted suggestion for now, I think adding these bound are good, but usually it requires updating tests, another bound we can add is `frame_system::Trait::Call: Dispatchable<Origin = Self::Origin>` but I didn't do so because it requires updating tests as well.

Reply:
Fair enough, I don't want this to block the PR.